### ISSUE TITLE
Use of Array.includes not flagged as error under ES5 lib + es3 target

### ISSUE THREAD

**Issue by GilbertSun** [microsoft/TypeScript#32405]
**TypeScript Version:**  3.5.3

**Search Terms:**

**Code**

```json
{
  "compilerOptions": {
    "lib": ["es5"],
    "target": "es3"
  }
}
```
```ts
const array = ['a', 'b', 'c'];
if (array.includes('a')) {
  document.write('right');
} else {
  document.write('wrong');
}
```

**Expected behavior:**
when i run `tsc`, i should got error 
1. Cannot find name 'document'. Do you need to change your target library? Try changing the `lib` compiler option to include 'dom'
2. some error tell me not to use includes because Array.prototype.includes is not include in es5

**Actual behavior:**
i only got the first error tell me not to use document and notice me to include dom in `lib`,
i did't get any error tell me not to use `Array.prototype.includes`, and my vscode not show any error

**Playground Link:** 

**Related Issues:**

**Comment by RyanCavanaugh** [comment]
I don't see this problem occurring.
```
a.ts:2:11 - error TS2339: Property 'includes' does not exist on type 'string[]'.

2 if (array.includes('a')) {
            ~~~~~~~~

a.ts:3:3 - error TS2584: Cannot find name 'document'. Do you need to change your target library? Try changing the `lib` compiler option to include 'dom'.

3   document.write('right');
    ~~~~~~~~

a.ts:5:3 - error TS2584: Cannot find name 'document'. Do you need to change your target library? Try changing the `lib` compiler option to include 'dom'.

5   document.write('wrong');
    ~~~~~~~~


Found 3 errors.
```

**Comment by GilbertSun** [comment]
sorry for wrong bug report, i found the reason why Array.includes not flagged as error under ES5 lib + es3 target, because my project has a `@types/node` in my node_modules, this lib auto import some .d.ts
by check the `typesVersions`

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/package.json#L5
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/ts3.2/index.d.ts#L9

```ts
// Reference required types from the default lib:
/// <reference lib="es2018" />
/// <reference lib="esnext.asynciterable" />
/// <reference lib="esnext.intl" />
/// <reference lib="esnext.bigint" />
```
i did't mean to add `@types/node` in my devDependencies, i was include by some other library, so, i think this problem may happend in many project, because of `@types/node`
